### ISSUE TITLE
Feature : replace hand-crafted auth things with Supabase Remix's package ?

### ISSUE THREAD

**Issue by rphlmr** [rphlmr/supa-fly-stack#54]
Hello there,

Someone ask this and maybe it's a good idea to not scatter people with multiple ways to handle auth in Remix with Supabase.

Maybe I should rewrite auth with https://supabase.com/docs/guides/auth/auth-helpers/remix.

###  I don't want to replace Prisma 🥶 but only the Auth part.
After that, it should be a breeze to replace Prisma with Supabase to query your database if this It what you need for your project.

I'll probably do some abstraction to keep helpers like `requireAuthSession` and keep things simple to use in protected loaders/actions.

Is It something you are interested in or do you prefer to be able to modify what you want with the actual auth implementation?

**Comment by rphlmr** [comment]
I’m waiting to be sure that the way it handles cookies (HttpOnly: false) feets my security requirements.

**Comment by timgit** [comment]
First of all, thanks for your work because it's very helpful. I used your stack to bootstrap my app, but I bypassed all prisma to make sure RLS policies work for my multi-tenancy use cases.  

Now that I'm wanting to upgrade to v2, I have a lot of work to do. :) One of the things that I'd love to see in the future is a way to pull in the latest changes from this stack without worrying I'm going to lose any changes.  For example, I extended AuthSession to add custom claims in the jwt metadata.

```js
export type AuthSession = {
    accessToken: string;
    refreshToken: string;
    userId: string;
    email: string;
    expiresIn: number;
    expiresAt: number;
    claims: {
      provider?: string
      [key: string]: any
    }
  }
```

**Comment by rphlmr** [comment]
Thanks @timgit ,

Unfortunately, Remix provides no way to upgrade a Stack after creating your app :(

For sure, I'm not the best to don't make any breaking changes as I change a lot of things recently, improving from my daily use of the code in this stack 🤣

I'll try to make more atomics commits to make it easy to follow.

I have to admit that this stack is not the best for RLS because I mainly use Supabase through Prisma (mainly for performance reasons).

I thought of doing an npm package with everything to handle auth and get a supabase client, but now there is the official Supabase auth package I don't know.

Basically, things that you may be interested in are : 
- [auth module](https://github.com/rphlmr/supa-fly-stack/tree/main/app/modules/auth)
- [user service](https://github.com/rphlmr/supa-fly-stack/tree/main/app/modules/user)
- [Supabase integration](https://github.com/rphlmr/supa-fly-stack/tree/main/app/integrations/supabase)

For your AuthSession extends, should should modify [this mapper](https://github.com/rphlmr/supa-fly-stack/blob/main/app/modules/auth/mappers.ts).
